Dockerfile

```diff
@@ -1,11 +1,21 @@
-ARG GOARCH="amd64"
-FROM golang:1.22 AS builder
+FROM --platform=$BUILDPLATFORM golang:1.22 AS builder
```

Comment:
None of these arguments need to be set manually. The build platform argument as used in the from line tells docker to always use the native architecture rather than to emulate with qemu as we use cross compilation. The target arguments get set based on the platform flag which is provided in the makefile.

https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/

Reply:
I knew I remember something about this but I didn't know what. 

Here is the problem, with multiplatform the `--load`flag fails

```
make image-build
docker buildx build . \
        --platform="linux/amd64,linux/arm64" \
        --tag="gcr.io/k8s-staging-networking/kube-network-policies:v20240527-abf99c2-dirty"
WARNING: No output specified with docker-container driver. Build result will only remain in the build cache. To push result image into registry use --push or to load image into docker use --load
[+] Building 33.9s (18/18) FINISHED             
```

This seems to solve the problem
```
diff --git a/Makefile b/Makefile
index 5067609..ade162c 100644
--- a/Makefile
+++ b/Makefile
@@ -40,8 +40,7 @@ PLATFORMS?=linux/amd64,linux/arm64
 image-build:
        docker buildx build . \
                --platform="${PLATFORMS}" \
-               --tag="${IMAGE}" \
-               --load
+               --tag="${IMAGE}"
```

Reply:
@vaskozl you resolved this comment but using the `--load` flag with multiple archs fails for me, is not failing for you?

Reply:
Sorry, I haven't resolved it since I the first comment. `--load` does indeed work for me. Apologies if it's still showing that way, I didn't mean to resolve it in the first place, github ate my comment and made me type it out twice earlier on my phone :/.

```
% make image-build
docker buildx build . \
		--platform="linux/amd64,linux/arm64" \
		--tag="gcr.io/k8s-staging-networking/kube-network-policies:v20240527-abf99c2" \
		--load
...
% docker images                            
REPOSITORY                                            TAG                 IMAGE ID       CREATED         SIZE
gcr.io/k8s-staging-networking/kube-network-policies   v20240527-abf99c2   04cd1b9d68b4   8 seconds ago   163MB```

It's not necessary though if one doesn't plan to use the image locally. Looks like it might not work on older version of `docker`, so I suggest:

* remove `--load` if we just want to test builds on all architectures without pushing on `image-build`
* if we want to load it to be able to test locally, we can instead leave `--load` and remove the `--platform` flag. It will only build for the host arch in that case.

What do you think?

Reply:
I've updated my commit to not build both platforms on `image-build` assuming the first option is preferable. 

(For reference my docker version is `26.0`)

Reply:
commented here https://github.com/kubernetes-sigs/kube-network-policies/pull/30/files#r1616335414 , defaulting to one platform is compatible

Reply:
> ii  docker-ce                                                                                         5:20.10.21~3-0~debian-bullseye                           amd64        Docker: the open-source application container engine

:(

Reply:
Would you not prefer to keep `PLATFORMS` set to the platforms we push for, and then work around the "local" limitation by not using `--platform` in `bulild-image` (only in `build-push`)?

That way one wouldn't need to set PLATFORMS even if working locally and `make image-push` would should push upstream without error?

Reply:
I say think because I personally think the fewer variables one has to juggle the better :D. I've already removed `--platform` from the local build. It's pretty redundant as you would never need a local alternate arch unless you ran that image with qemu.

Reply:
I'm not an expert here, I trust you